### ISSUE TITLE
Support for dynamichttp

### ISSUE THREAD

**Issue by GKNSB** [MythicAgents/Apollo#79]
Hey,

Are there any plans to support dynamichttp as c2 profile? Being able to craft the communication would be really good.

**Comment by djhohnstein** [comment]
Yes, there is plans to do so. It's in the backlog of things to do as I become available and have time to do so. It should be a relatively easy thing to add as all the core HTTP code is there already. If you wanted to contribute, there are some docs on the agent documentation page of the Mythic instance you spin up as well.

**Comment by GKNSB** [comment]
Yeap to be honest I thought about starting to work on it myself and contribute. I'll see if I find the time this weekend to become more familiar with the structure and the code, and I might be able to cook something up. It's good to hear there are plans to support it though in the case I don't find the time to mess with it.

**Comment by djhohnstein** [comment]
Sounds good! If not, no worries. It's on my timeline.

**Comment by cr7pt0pl4gu3** [comment]
Hey, what's the status for dynamichttp and are there plans to implement it in near future?
Considering to use this agent but this is the main issue for now.

**Comment by djhohnstein** [comment]
Currently what my blocker is my development VM died in my lab, and I just haven't taken the time to look into what the issue is to recover it as I've been working on some other projects. I don't have an ETA on that but I have another project I need to retrieve from the VM, so I'd imagine I'll have the VM recovered soon enough.

Ultimately, if someone wanted to contribute, you'd create a new project like `DynamicHttpProfile` and essentially copy-paste [this file](https://github.com/MythicAgents/Apollo/blob/master/Payload_Type/apollo/agent_code/HttpProfile/HttpProfile.cs) and modify lines [143-176](https://github.com/MythicAgents/Apollo/blob/master/Payload_Type/apollo/agent_code/HttpProfile/HttpProfile.cs)

**Comment by cr7pt0pl4gu3** [comment]
Got it, thanks for the update!

**Comment by djhohnstein** [comment]
Just rounding up - I checked on the VM this week again to see if I could fix it, but it's big time broken. I'm getting more gas in the tank as of late so I'll look at knocking some stuff out here soonish.

**Comment by its-a-feature** [comment]
closing this in favor of supporting `httpx` going forward
